ca/ecdsa_allow_list.go

```diff
@@ -67,17 +34,23 @@ func makeRegIDsMap(regIDs []int64) map[int64]bool {
 	return regIDsMap
 }
 
-// NewECDSAAllowListFromFile is exported to allow `boulder-ca` to
-// construct a new `ECDSAAllowList` object. An initial entry count is
-// returned to `boulder-ca` for logging purposes.
-func NewECDSAAllowListFromFile(filename string, logger log.Logger, metric *prometheus.GaugeVec) (*ECDSAAllowList, int, error) {
-	allowList := &ECDSAAllowList{logger: logger}
-	// Create an allow list reloader. This also populates the inner
-	// allowList regIDsMap.
-	reloader, err := reloader.New(filename, allowList.Update, logger)
+// NewECDSAAllowListFromFile is exported to allow `boulder-ca` to construct a
+// new `ECDSAAllowList` object. It returns the ECDSAAllowList, the size of allow
+// list after attempting to load it (for CA logging purposes so length() doesn't
+// need to be exposed), or an error.
+func NewECDSAAllowListFromFile(filename string) (*ECDSAAllowList, int, error) {
+	configBytes, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	var regIDs []int64
+	err = strictyaml.Unmarshal(configBytes, &regIDs)
 	if err != nil {
 		return nil, 0, err
 	}
-	allowList.reloader = reloader
+	allowList := &ECDSAAllowList{}
+	allowList.regIDsMap = makeRegIDsMap(regIDs)
+
 	return allowList, allowList.length(), nil
```

Comment:
```suggestion
	return allowList, len(allowList), nil
```

Reply:
My mistake here, len() the inner mapping.

Reply:
Already done. Good idea.